cmd/blockchaincmd/add_validator.go

```diff
@@ -264,54 +267,35 @@ func addValidator(cmd *cobra.Command, args []string) error {
 	subnetID := sc.Networks[network.Name()].SubnetID
 
 	// if user chose to upsize a local node to add another local validator
+	var localValidatorClusterName string
 	if createLocalValidator {
-		anrSettings := node.ANRSettings{}
-		nodeConfig := map[string]interface{}{}
-		ux.Logger.PrintToUser("Creating a new Avalanche node on local machine to add as a new validator to blockchain %s", blockchainName)
-		if app.AvagoNodeConfigExists(blockchainName) {
-			nodeConfig, err = utils.ReadJSON(app.GetAvagoNodeConfigPath(blockchainName))
-			if err != nil {
-				return err
-			}
-		}
-		if partialSync {
-			nodeConfig[config.PartialSyncPrimaryNetworkKey] = true
-		}
-		avalancheGoBinPath, err := node.GetLocalNodeAvalancheGoBinPath()
+		// TODO: make this to work even if there is no local cluster for the blockchain and network
+		targetClusters, err := localnet.GetFilteredLocalClusters(app, true, network, blockchainName)
 		if err != nil {
-			return fmt.Errorf("failed to get local node avalanche go bin path: %w", err)
-		}
-
-		nodeName := ""
-		blockchainID := sc.Networks[network.Name()].BlockchainID
-
-		if nodeName, err = node.UpsizeLocalNode(
-			app,
-			network,
-			blockchainName,
-			blockchainID,
-			subnetID,
-			avalancheGoBinPath,
-			nodeConfig,
-			anrSettings,
-		); err != nil {
 			return err
 		}
-		// get node data
-		nodeInfo, err := node.GetNodeInfo(nodeName)
+		if len(targetClusters) == 0 {
+			return fmt.Errorf("no local cluster is running for network %s and blockchain %s", network.Name(), blockchainName)
+		}
+		if len(targetClusters) != 1 {
+			return fmt.Errorf("too many local clusters running for network %s and blockchain %s", network.Name(), blockchainName)
+		}
+		localValidatorClusterName = targetClusters[0]
+		node, err := localnet.AddNodeToLocalCluster(app, ux.Logger.PrintToUser, localValidatorClusterName, httpPort, stakingPort)
 		if err != nil {
 			return err
 		}
-		nodeIDStr, publicKey, pop, err = node.GetNodeData(nodeInfo.Uri)
+		nodeIDStr, publicKey, pop, err = utils.GetNodeID(node.URI)
 		if err != nil {
 			return err
 		}
-		// update sidecar with new node
-		if err := node.AddNodeInfoToSidecar(&sc, nodeInfo, network); err != nil {
+		if err := app.AddDefaultBlockchainRPCsToSidecar(blockchainName, network, []string{node.URI}); err != nil {
```

Comment:
why not also add &sc as argument in AddDefaultBlockchainRPCsToSidecar so that there is no need to reload

Reply:
that is going to assume the sidecar is always going to be already loaded, which may not be the case in general.
 I can make this function to return the new sidecar